### PR TITLE
Added Angular SDK examples

### PR DESCRIPTION
**Pull request by engineering-this**
## CKEditor 4 Angular SDK examples
- Editor types, which presents `divarea` and `inline` editors.
- Component events and component attribute directives.
- Form example, with presentation of two-way binding.

## Todo
- Improve styling.
- Refactor styles to classes.
- Move styles to static file, so few deps could be dropped and webpack would run faster.

For now I didn't focus much on styling, as I'd rather hear out other people opinion.

## How to run
For some reason angular package doesn't work with linked package, so we need to create package locally and install it. Preserving symlinks in webpack doesn't solve it.

- You need to clone angular repo, build it and pack it:
```npm run build-package```
Inside `dist` run:
```npm pack```
- In `ckeditor-docs` install new deps from this branch.
- Install packed package:
```npm install location/to/built/tgz```

## Additional notes
Webpack doesn't work when two webpack config files are passed to grunt. After adding working webpack config to grunt both React and Angular build tasks doesn't work. As a workaround this I've added npm commands to build each integration, and forced grunt to use them.

This is second part to https://github.com/ckeditor/ckeditor-dev/issues/2920.

Closes ckeditor/ckeditor4-angular#14. Closes ckeditor/ckeditor-dev#2920.

### CODE REVIEW

File: gruntfile.js [hunk 1]
```diff
@@ -66,9 +65,27 @@ module.exports = function( grunt ) {
 		done();
 	} );
 
-	grunt.registerTask( 'build-react', [ 'webpack:react' ] );
-	grunt.registerTask( 'docs', [ 'api', 'fix-scayt-docs', 'prepare-examples', 'build-react', 'umberto' ] );
-	grunt.registerTask( 'docs-serve', [ 'api', 'fix-scayt-docs', 'prepare-examples', 'build-react', 'umberto', 'connect' ] );
+
+	function buildIntegrationTask( name ) {
```
Comment (f1ames) [inline gruntfile.js]: Could you move this function to the end of the file?

**Comment by engineering-this** [comment]
New changes:

- rebased,
- moved integrations into `scripts/integrations`,
- enabled IE polyfills,
- removed `dirty` and `touched` messages,
- moved styles to static html file and removed `.css` files from Angular source, co they aren't bundled by webpack, that allowed to remove some deps,
- updated styles (borrowed from React sample).

**Comment by f1ames** [comment]
Rebased on latest `master`.

File: scripts/integrations/angular/src/app/editor-types/editor-types.component.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+@Component( {
+	selector: 'app-editor-types',
+	templateUrl: './editor-types.component.html'
+} )
+export class EditorTypesComponent {
+	public isReadOnly = false;
+	public editorData =
+		`<p>Hello world!</p>`;
+
+	inline = false;
```
Comment (ma2ciek) [inline scripts/integrations/angular/src/app/editor-types/editor-types.component.ts]:
```suggestion
	public inline = false;
```

File: scripts/integrations/angular/src/app/events/events.component.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+@Component( {
+	selector: 'app-events',
+	templateUrl: './events.component.html'
+} )
+export class EventsComponent {
+	public isReadOnly = false;
+
+	isHidden = false;
```
Comment (ma2ciek) [inline scripts/integrations/angular/src/app/events/events.component.ts]: Also, I'd mark these props as private/public or I'd not mark anything at all.

File: scripts/integrations/angular/src/app/events/events.component.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+	isRemoved = false;
+
+	inline = false;
+
+	public componentEvents: ComponentEvent[] = [];
+
+	clearEventsLog() {
```
Comment (ma2ciek) [inline scripts/integrations/angular/src/app/events/events.component.ts]: Also, this method is inconsistent with the next ones in the return type annotation.

File: scripts/integrations/angular/src/app/events/events.component.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+		this.componentEvents.length = 0;
+	}
+
+	saveEvent( eventName: string ): void {
```
Comment (ma2ciek) [inline scripts/integrations/angular/src/app/events/events.component.ts]: This method can be marked as private and moved to the bottom of this file.

File: scripts/integrations/angular/src/app/events/events.component.ts (hunk 3 above)
Comment (ma2ciek) [reply scripts/integrations/angular/src/app/events/events.component.ts]: The rule can be enforced by the TSLint.

**Comment by ma2ciek** [comment]
I added a few small comments to the TS files. We have not any style convention for them, but it would be good to have them styled the same way across the project.

**Comment by ma2ciek** [comment]
And one thing which I discovered today is that in the CKEditor 5 integrations we have a few issues with multiple editors created with usage of the `*ngFor` directive - https://github.com/ckeditor/ckeditor5-angular/issues/86 - it'd be good to check it in the CKEditor 4 integration too (but for me it can be done in a follow-up).

**Comment by f1ames** [comment]
@AnnaTomanek could you take a final look on this PR?

**Comment by AnnaTomanek** [comment]
I added some corrections, I feel some stuff could still be improved.

The Events example contains no explanation text -- see the React sample for comparison/inspiration. Same for the Form example.

(minor) Some spacing could be added between the log and the button.

**Comment by f1ames** [comment]
@engineering-this could you take a look at https://github.com/ckeditor/ckeditor-docs/pull/223#issuecomment-486622581 and try to add some descriptions for examples as suggested?

**Comment by engineering-this** [comment]
Added short descriptions to samples.

**Comment by f1ames** [comment]
Rebased on the latest `master`.

**Comment by f1ames** [comment]
Some minor rewording.

@engineering-this The `@Input` is mentioned (as a link anchor) in the [description here](https://github.com/ckeditor/ckeditor-docs/pull/223/files#diff-3d09f6c821ff3ba42007d2eb4dfb5f04R4), however link points to `@Output` properties. Was it intended or just a typo?

**Comment by engineering-this** [comment]
Just a silly mistake. I just fixed it.

**Comment by f1ames** [comment]
Rebased on the latest `master`.

**Pull request merged**